File: Jack.Presentation.Web/Controllers/API/PresencaController.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Net;
using System.Web.Http;

namespace Controllers.API
{
	public class PresencaController : ApiController
	{

		[HttpGet()]
		public IList<Model.Familia> GetValue(int ID)
		{

			List<Model.Familia> lstRetorno = null;
			Business.Presenca oBusiness = default(Business.Presenca);

			try {
				oBusiness = new Business.Presenca();
				lstRetorno = oBusiness.Load(ID);

			} catch (Exception ex) {
				lstRetorno = null;
			} finally {
				oBusiness = null;
			}

			return lstRetorno;
		}

		public List<Model.FamiliaPresenca> ObterPresencaPorMae(int Familia, int Ano)
		{

			List<Model.FamiliaPresenca> lstRetorno = null;
			Business.Presenca oBusiness = default(Business.Presenca);

			try {
				oBusiness = new Business.Presenca();
				lstRetorno = oBusiness.ObterPresencaPorMae(Familia, Ano);

			} catch (Exception ex) {
				lstRetorno = null;
			} finally {
				oBusiness = null;
			}

			return lstRetorno;
		}

	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================

File: Jack.Presentation/Controllers/API/PresencaController.vb
Imports System.Net
Imports System.Web.Http

Namespace Controllers.API
    Public Class PresencaController
        Inherits ApiController

        <HttpGet>
        Public Function GetValue(ID As Integer) As IList(Of Model.Familia)

            Dim lstRetorno As List(Of Model.Familia) = Nothing
            Dim oBusiness As Business.Presenca

            Try
                oBusiness = New Business.Presenca()
                lstRetorno = oBusiness.Load(ID)

            Catch ex As Exception
                lstRetorno = Nothing
            Finally
                oBusiness = Nothing
            End Try

            Return lstRetorno
        End Function

        Public Function ObterPresencaPorMae(Familia As Integer, Ano As Integer) As List(Of FamiliaPresenca)

            Dim lstRetorno As List(Of FamiliaPresenca) = Nothing
            Dim oBusiness As IPresencaApp

            Try
                oBusiness = New PresencaApp()
                lstRetorno = oBusiness.ObterPresencaPorMae(Familia, Ano)

            Catch ex As Exception
                lstRetorno = Nothing
            Finally
                oBusiness = Nothing
            End Try

            Return lstRetorno
        End Function

    End Class
End Namespace